articles/compliance/gdpr/features-aiding-compliance/user-consent.md

```diff
@@ -0,0 +1,353 @@
+
+```har
+{
+    "method": "GET",
+    "url": "https://${account.namespace}/api/v2/users/YOUR_USER_ID",
+    "httpVersion": "HTTP/1.1",
+    "headers": [{
+        "name": "Authorization",
+        "value": "Bearer YOUR_MGMT_API_ACCESS_TOKEN"
+    }],
+    "queryString":  [
+        {
+          "name": "fields",
+          "value": "user_metadata"
+        }
+    ]
+}
+```
+
+Sample response:
+
+```json
+{
+  "user_metadata": {
+    "consent": {
+	    "given": true,
+	    "date": "01/23/2018",
+	    "text_details": "some-url"
+  	}
+  }
+}
+```
+
+### Search for a set of users
+
+To search for a set of users, use [the List or search users endpoint](/users/search#users). 
+
+This endpoint is eventually consistent (that is, the response might not reflect the results of a recently-complete write operation) and that [only specific fields are available for search](/api/management/v2/user-search#searchable-fields). 
+
+Consent information that are saved as part of the `user_metadata` are not searchable. 
+
+For a sample request and response see [Search Users](/users/search#users). For more examples, see [Example Queries](/api/management/v2/user-search#example-queries).
+
+### Update consent information
+
+To update a user's `user_metadata`, use [the Update a user endpoint](/api/management/v2#!/Users/patch_users_by_id).
+
+How you will structure your request depends on how you have structured your metadata: as root or as inner properties.
+
+Metadata as root properties:
+
+```json
+{
+  "consentGiven": true,
+  "consentDetails": "some-url"
+}
+```
+
+Metadata as inner properties:
+
+```json
+{
+  "consent": {
+    "given": true,
+    "text_details": "some-url"
+  }
+}
+```
+
+#### Update a root property
+
+In this case the metadata are merged so you need only send the field you want to update. For example, let's say we want to add a consent date and set it to `01/23/2018`.
+
+```har
+{
+    "method": "PATCH",
+    "url": "https://${account.namespace}/api/v2/users/USER_ID",
+    "httpVersion": "HTTP/1.1",
+    "headers": [{
+        "name": "Authorization",
+        "value": "Bearer YOUR_MGMT_API_ACCESS_TOKEN"
+    },
+    {
+        "name": "content-type",
+        "value": "application/json"
+    }],
+    "postData" : {
+        "mimeType": "application/json",
+        "text": "{\"user_metadata\":{\"consentDate\":\"01/24/2018\"}}"
+    }
+}
+```
+
+This will add a new property to the user profile, the **user_metadata.consentDate**, which will hold the date the customer consented. The response will be the full user profile. The updated metadata will look like this:
+
+```json
+{
+  "consentGiven": true,
+  "consentDate": "01/23/2018",
+  "consentDetails": "some-url"
+}
+```
+
+#### Update an inner property
+
+In this case you have to send the whole object, even though you might want to update only one property. If you do not, the other properies will be removed. Let's add an inner property for the consent date and set it to `01/23/2018`.
```

Comment:
To update an inner property, you must send the whole metadata object, even if you aren't updating more than one property. **If you do not include the entire object, Auth0 will remove your existing properties.**